store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java

```diff
@@ -129,19 +136,17 @@ public void service(HttpServletRequest req, HttpServletResponse resp) throws IOE
      */
     public static void unregister(ZimbraExtension ext) {
         synchronized(sHandlers) {
-            for (Iterator it = sHandlers.keySet().iterator(); it.hasNext(); ) {
-                String path = (String) it.next();
-                if (path.startsWith("/" + ext.getName())) {
-                    ExtensionHttpHandler handler = (ExtensionHttpHandler) sHandlers.get(path);
-                    try {
-                        handler.destroy();
-                    } catch (Exception e) {
-                        ZimbraLog.extensions.warn("Error in destroy for handler " + handler.getClass(), e);
-                    } finally {
-                        it.remove();
-                    }
+            String extName = ext.getName();
+            for (ExtensionHttpHandler extHandler: EXTENSION_HANDLERS.get(extName)) {
```

Comment:
why for loop, are there more extensions with same name?

Reply:
Loop is for multiple handlers for extension.